### ISSUE TITLE
Group Invitation relationship not deleted when an external user joins the site (automatically joins the group)

### ISSUE THREAD

**Issue by oseg** [ColdTrick/group_tools#128]
Context: elgg v2.3.3 + group_tools v9.2.2

Scenario:
- In group tools plugin settings, set 'Allow all users to be invited by e-mail address' to 'yes'
- On 'mygroup' page, click on the 'invite users' button
- Click on 'using e-mail address' tab, enter a valid email address and press 'invite' button
- the 'new user' receives an invitation email, but did not follow the invitation link to create his account
- He creates his elgg account by himself, directly on the site register page
- At his 1st connection, his dashboard is displayed
- The 'group membership' widget displays 'mygroup'

==> But at the same time, the "group invitation" widget displays 'mygroup', if the user clicks on the 'accept' button, an error message saying that he is already a member of that group is displayed

**Comment by jeabakker** [comment]
the new user registers with the e-mail address he is invited on?

**Comment by oseg** [comment]
No, he did not follow the register link contained in the email (the one with the group_invitecode).

He visited the site and found the register page by himself, he then registered (without the group_invitecode).

**Comment by jeabakker** [comment]
> No, he did not follow the register link contained in the email (the one with the group_invitecode).
> 
> He visited the site and found the register page by himself, he then registered (without the group_invitecode).

That i understood, but does the user register with the same e-mail address as that is used to send out the invite?

So 
- group owner invites user with e-mail user@example.com
- the user gets an email, ignores the mail
- user goes to the site
- clicks the register link
- fills out the form and for email address enters user@example.com??

**Comment by oseg** [comment]
yes
